packages/app-file-manager/src/components/BulkActions/ActionEdit/ActionEdit.tsx

```diff
@@ -0,0 +1,134 @@
+
+export const ActionEdit = observer(() => {
+    const { fields: defaultFields } = useFileModel();
+    const {
+        useWorker,
+        useButtons,
+        useDialog: useBulkActionDialog
+    } = FileManagerViewConfig.Browser.BulkAction;
+    const worker = useWorker();
+    const { updateFile } = useFileManagerView();
+    const { canEdit } = useFileManagerApi();
+    const { IconButton } = useButtons();
+    const { showConfirmationDialog, showResultsDialog } = useBulkActionDialog();
+
+    const presenter = useMemo<ActionEditPresenter>(() => {
+        return new ActionEditPresenter();
+    }, []);
+
+    useEffect(() => {
+        presenter.load(defaultFields);
+    }, [defaultFields]);
+
+    const filesLabel = useMemo(() => {
+        return getFilesLabel(worker.items.length);
+    }, [worker.items.length]);
+
+    const canEditAll = useMemo(() => {
+        return worker.items.every(item => canEdit(item));
+    }, [worker.items]);
+
+    const openWorkerDialog = (batch: BatchDTO) => {
+        showConfirmationDialog({
+            title: "Edit files",
+            message: `You are about to edit ${filesLabel}. Are you sure you want to continue?`,
+            loadingLabel: `Processing ${filesLabel}`,
+            execute: async () => {
+                await worker.processInSeries(async ({ item, report }) => {
+                    try {
+                        const extensions = defaultFields.find(
+                            field => field.fieldId === "extensions"
+                        );
+
+                        const extensionsData = GraphQLInputMapper.toGraphQLExtensions(
+                            item.extensions,
+                            batch
+                        );
+
+                        const output = omit(item, ["id", "createdBy", "createdOn", "src"]);
+
+                        const fileData = {
+                            ...output,
+                            extensions: prepareFormData(
+                                extensionsData,
+                                extensions?.settings?.fields || []
+                            )
+                        };
+
+                        await updateFile(item.id, fileData);
+
+                        report.success({
+                            title: `${item.name}`,
+                            message: "File successfully edited."
+                        });
+                    } catch (e) {
+                        report.error({
+                            title: `${item.name}`,
+                            message: e.message
+                        });
+                    }
+                });
+
+                worker.resetItems();
+
+                showResultsDialog({
+                    results: worker.results,
+                    title: "Edit files",
+                    message: "Finished editing files! See full report below:"
+                });
+            }
+        });
+    };
```

Comment:
Just a comment on this: I think it would be great to create a generic `DialogPresenter` class which can be reused across other presenter implementations, to manage thee dialog interactions. That way we'll be able to remove this huge bloat of coad from the components. Something to try.